extensions/metrics/src/interceptors/metrics.interceptor.ts

```diff
@@ -93,21 +92,31 @@ export class MetricsInterceptor implements Provider<Interceptor> {
     next: () => ValueOrPromise<T>,
   ) {
     MetricsInterceptor.setup();
-    const {request, response} = invocationCtx.parent as RequestContext;
+    const {source, parent} = invocationCtx;
     const labelValues: LabelValues<LabelNames> = {
       targetName: invocationCtx.targetName,
-      method: request.method,
     };
-    labelValues.path = getPathPattern(invocationCtx, request);
+    if (isRouteSource(source)) {
+      // parent context will be request context if invocation source is route
+      labelValues.method = (parent as RequestContext).request.method;
+      labelValues.path = getPathPattern(source);
```

Comment:
The RouteSource already has `verb` property - https://github.com/strongloop/loopback-next/blob/7f764e109a23da5e03f9a0bee93ba0c7c7d21e81/packages/rest/src/router/route-entry.ts#L17. You don't need to get the http method from `parent`.